### PR TITLE
TASK-28152:Admin user can access documents of spaces where he's not member

### PR DESCRIPTION
**Pull request by hassenelkamel-exo**

### CODE REVIEW

File: data-upgrade-ecms/src/main/resources/conf/portal/configuration.xml [hunk 1]
```diff
@@ -82,6 +82,40 @@
...
+        </value-param>
+        <value-param>
+          <name>plugin.upgrade.async.execution</name>
+          <description>Execute this upgrade asynchronously</description>
+          <value>true</value>
+        </value-param>
+        <value-param>
+          <name>plugin.upgrade.target.version</name>
+          <description>Target version of the plugin</description>
+          <value>6.2.0</value>
```
Comment (plamarque) [inline data-upgrade-ecms/src/main/resources/conf/portal/configuration.xml:115]: the plugin will only be triggered for 6.2.0 version ? If so, how are we going to deal with continuous delivery and milestone deploymnents?

File: data-upgrade-ecms/src/main/java/org/exoplatform/ecms/upgrade/templates/DocumentPermissionUpgradePlugin.java [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+    public void processUpgrade(String oldVersion, String newVersion) {
+        long startupTime = System.currentTimeMillis();
+        log.info("Start upgrade of space documents jcr nodes");
+
+        SessionProvider sessionProvider = null;
+        try {
+            sessionProvider = sessionProviderService.getSystemSessionProvider(null);
+            ManageableRepository repository = repositoryService.getCurrentRepository();
+            Session session = sessionProvider.getSession(repository.getConfiguration().getDefaultWorkspaceName(), repository);
+            String spacesNodePath = nodeHierarchyCreator.getJcrPath(BasePath.CMS_GROUPS_PATH) + DOCUMENTS_NODE;
```
Comment (ahamdi) [inline data-upgrade-ecms/src/main/java/org/exoplatform/ecms/upgrade/templates/DocumentPermissionUpgradePlugin.java]:
```suggestion
            String spacesNodePath = nodeHierarchyCreator.getJcrPath(BasePath.CMS_GROUPS_PATH) + "/spaces/;
```

File: data-upgrade-ecms/src/main/java/org/exoplatform/ecms/upgrade/templates/DocumentPermissionUpgradePlugin.java [hunk 3]
```diff
@@ -0,0 +1,79 @@
...
+            Node spacesRootNode = (Node) session.getItem(spacesNodePath);
+            if (!spacesRootNode.isNodeType(EXO_PRIVILEGEABLE)) {
+                spacesRootNode.addMixin(EXO_PRIVILEGEABLE);
+            }
```
Comment (ahamdi) [inline data-upgrade-ecms/src/main/java/org/exoplatform/ecms/upgrade/templates/DocumentPermissionUpgradePlugin.java]:
```suggestion

```

File: data-upgrade-ecms/src/main/java/org/exoplatform/ecms/upgrade/templates/DocumentPermissionUpgradePlugin.java [hunk 4]
```diff
@@ -0,0 +1,79 @@
...
+            NodeIterator iter = spacesRootNode.getNodes();
+            while (iter.hasNext()) {
+                Node spaceNode = iter.nextNode();
+                if (spaceNode!= null) {
+                    ((ExtendedNode) spaceNode).removePermission(ADMINISTRATORS_IDENTITY);
+                    spaceNode.save();
+                }
```
Comment (ahamdi) [inline data-upgrade-ecms/src/main/java/org/exoplatform/ecms/upgrade/templates/DocumentPermissionUpgradePlugin.java]:
```suggestion
                Node spaceNode = iter.nextNode();
                            Node spaceDocumentsFolder = spaceNode.getNode("Documents");
            if (!spaceDocumentsFolder.isNodeType(EXO_PRIVILEGEABLE)) {
                spaceDocumentsFolder.addMixin(EXO_PRIVILEGEABLE);
            }
                  ((ExtendedNode) spaceDocumentsFolder).removePermission(ADMINISTRATORS_IDENTITY);
                  spaceDocumentsFolder.save();
```

**Pull request closed without merging**